pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/SingularEmailEventUpdates.java

```diff
@@ -0,0 +1,107 @@
+        if (!studyCfg.getString("study.guid").equals(STUDY_GUID)) {
+            throw new DDPException("This task is only for the " + STUDY_GUID + " study!");
+        }
+        File file = cfgPath.getParent().resolve(DATA_FILE).toFile();
+        if (!file.exists()) {
+            throw new DDPException("Data file is missing: " + file);
+        }
+        this.dataCfg = ConfigFactory.parseFile(file).resolveWith(varsCfg);
+        this.cfgPath = cfgPath;
+        this.cfg = studyCfg;
+        this.varsCfg = varsCfg;
+    }
+
+    @Override
+    public void run(Handle handle) {
+        log.info("TASK:: SingularEmailEventUpdates ");
+        StudyDto studyDto = handle.attach(JdbiUmbrellaStudy.class).findByStudyGuid(cfg.getString("study.guid"));
+        updateEmailEvents(handle, studyDto);
+    }
+
+    private void updateNotificationAction(Handle handle, StudyDto studyDto, String templateKey, String currDelaySeconds,
+                                          String newDelaySeconds, SingularEmailEventUpdates.SqlHelper helper) {
+        long notificationTemplateId = handle.attach(EventActionSql.class)
+                .findNotificationTemplate(templateKey, "en")
+                .map(NotificationTemplate::getId)
+                .get();
+
+        //update postDelaySeconds
+        long currPostDelaySeconds = Long.valueOf(currDelaySeconds);
+        long newPostDelaySeconds = Long.valueOf(newDelaySeconds);
+        long eventConfigId = helper.findNotificationEventConfigId(studyDto.getId(), notificationTemplateId, currPostDelaySeconds);
+        int rowCount = helper.updateDelayToEvent(eventConfigId, newPostDelaySeconds);
+        if (rowCount != 1) {
+            throw new DDPException("Updated : " + rowCount + " event configurations !!" );
+        }
+        log.info("Updated postDelaySeconds for event config ID : {} ", eventConfigId);
+    }
+
+    private void updateEmailEvents(Handle handle, StudyDto studyDto) {
+        var helper = handle.attach(SingularEmailEventUpdates.SqlHelper.class);
+        if (!dataCfg.hasPath("emailEvents")) {
+            throw new DDPException("There is no 'emailEvents' configuration.");
+        }
+        log.info("Updating singular email events configuration postDelaySeconds...");
+        List<? extends Config> events = dataCfg.getConfigList("emailEvents");
+        for (Config eventCfg : events) {
+            log.info("Updating event configuration postDelaySeconds for singular email : {}", eventCfg.getString("key"));
+            String templateKey = varsCfg.getString("emails." + eventCfg.getString("key"));
+            updateNotificationAction(handle, studyDto, templateKey,
+                    eventCfg.getString("currPostDelaySeconds"), eventCfg.getString("newPostDelaySeconds"), helper);
+        }
+        log.info("Email Event configurations (postDelaySeconds) has been updated in study {}", cfg.getString("study.guid"));
+    }
+
+    private interface SqlHelper extends SqlObject {
```

Comment:
What do you think about moving those methods to the separate DAO class EventConfigurationDao?

Reply:
Thats a good one to do.
we can do that as part of broader works where we need to look at how we can improve existing study config updates.
Go through all patches and see how we can develop some reusable patches